scalyr_agent/monitor_utils/k8s.py

```diff
@@ -2400,6 +2401,138 @@ def query_stats(self):
+
+    def _check_match(self, element, name, value, glob):
+        """
+        Checks to see if we have a match against the glob for a certain value
+        @param element: The index number of the element in the k8s_config list
+        @param name: A string containing the name of the field we are evaluating (used for debug log purposes)
+        @param value: The value of the field to evaluate the glob against
+        @param glob: A string containing a glob to evaluate
+        """
+        result = False
+        if glob is not None and value is not None:
+            # ignore this config if value doesn't match the glob
+            if fnmatch.fnmatch(value, glob):
+                result = True
+            else:
+                self._logger.log(
+                    scalyr_logging.DEBUG_LEVEL_2,
+                    "Ignoring k8s_log item %d because %s '%s' doesn't match '%s'"
+                    % (element, name, value, glob,),
+                )
+        return result
+
+    def get_log_config(
+        self, info, k8s_info, container_attributes, parser, rename_logfile,
+    ):
+        """
+        Creates a log_config from various attributes and then applies any `k8s_logs` configs that
+        might apply to this log
+        @param info: A dict containing docker information about the container we are creating a config for
+        @param k8s_info: A dict containing k8s information about hte container we are creating a config for
+        @param container_attributes: A set of attributes to add to the log config of this container
+        @param parser: A string containing the name of the parser to use for this log config
+        @param rename_logfile: A string containing the name to use for the renamed log file
+        @return: A dict containing a log_config, or None if we couldn't create a valid config (i.e. log_path was empty)
+        """
+
+        # Make sure we have a log_path for the log config
+        path = info.get("log_path", None)
+        if not path:
+            return None
+
+        # Build up the default config that we will use
+        result = {
+            "parser": parser,
+            "path": path,
+            "parse_format": self.__parse_format,
+            "attributes": container_attributes,
+            "rename_logfile": rename_logfile,
+            "rename_no_original": self.__rename_no_original,
+        }
+
+        # If we don't have any k8s information then we don't match against k8s_log_configs
+        if k8s_info is None:
+            return result
+
+        # Now apply log configs
+        for i, config in enumerate(self.__k8s_log_configs):
+            # We check for glob matches against `k8s_pod_glob`, `k8s_namespace_glob` and `k8s_container_glob`
+
+            # Check for the pod glob
+            pod_glob = config.get("k8s_pod_glob", None)
+            pod_name = k8s_info.get("pod_name", None)
+            if not self._check_match(i, "pod_name", pod_name, pod_glob):
```

Comment:
How about modifying `_check_match` to take in both the `k8s_info` and `config` objects, and then just pass in `k8s_pod_glob`.  That way you could do the gets for the `pod_glob` and `pod_name` in that method.  It would mean just passing in one more argument that you are now, but then you can eliminate 6 duplicate lines in this code block.  You can then also then write this block to be a single `if` statement:

```
  if self._check_match(i, "pod_name", ... ) or
    self._check_match(i, "k8s_container_name", ...) or
    self._check_match(i, "pod_namespace", ...):
    # match found.
```

Avoiding the sprinkle of `continue` statements throughout a body like this usually is better for readability.

Reply:
The main reason is because I also wanted those values for use in the following debug statement saying that we passed.

Reply:
Ah, right... I missed that part.  Ok, never mind about my suggestion.